src/Advection/vector_invariant_advection.jl

```diff
@@ -124,7 +124,7 @@ Adapt.adapt_structure(to, scheme::VectorInvariant{N, FT}) where {N, FT} =
 const VectorInvariantEnergyConserving    = VectorInvariant{<:Any, <:Any, <:EnergyConservingScheme}
 const VectorInvariantEnstrophyConserving = VectorInvariant{<:Any, <:Any, <:EnstrophyConservingScheme}
 
-const VectorInvariantConserving = Union{VectorInvariantEnergyConserving, VectorInvariantEnstrophyConserving}
+const VectorInvariantVerticallyConserving  = VectorInvariant{<:Any, <:Any, <:Any, Nothing, <:Any, <:Any, <:EnergyConservingScheme}
```

Comment:
what does VerticallyConserving mean?

Reply:
A conservative scheme in the vertical direction

Reply:
The other form also conserves momentum right? The difference is not whether they conserve momentum but whether we are using the "conservative form" (eg with this semantics, the "other" scheme is also "conserving"). How about `VerticallyConservativeVectorInvariant`?

Reply:
Sorry, i meant conservation of energy not momentum. `VerticallyConserving` conserves energy, while all the others do not because, to conserve energy, the discretization of `wdu` has to be consistent with the discretization of `dK`. I can change it to `VerticallyEnergyConserving`

Reply:
Yes do that!